ykpack/src/types.rs

```diff
@@ -892,6 +899,12 @@ pub enum Terminator {
         /// The return value and basic block to continue at, if the call converges.
         destination: Option<(IPlace, BasicBlockIndex)>,
     },
+    /// A call to yktrace::trace_debug. This is converted into a Statement::Debug at TIR
+    /// compilation time.
+    TraceDebugCall {
+        msg: String,
+        destination: BasicBlockIndex,
```

Comment:
What's the point of `destination`? We never seem to use it.

Reply:
You're right that we don't. I included it because I hope that one day we will add a verification step to the TIR compiler which checks that the next location is possible according to the SIR control flow.

We can kill this for now if you like though.

Reply:
That's fine then. Leave it in.